.github/workflows/source-and-docs-release.yml

```diff
@@ -0,0 +1,100 @@
+on:
+  workflow_dispatch:
+    inputs:
+      git_remote:
+        type: choice
+        description: "Git remote to checkout"
+        options:
+          - python
+          - Yhg1s
+          - pablogsal
+          - ambv
+      git_commit:
+        type: string
+        description: "Git commit to target for the release. Must use the full commit SHA, not the short ID"
+      cpython_release:
+        type: string
+        description: "CPython release number (ie '3.11.5', note without the 'v' prefix)"
+
+name: "Build Python source and docs artifacts"
+
+jobs:
+  source-and-docs:
```

Comment:
Would like to add a GitHub Environment to allow staging and reviewing an execution w/ approvals. I can create a separate GitHub issue for this if it's desirable.

Reply:
We have this for Windows installers so yeah, that would be nice!